tfjs-backend-webgpu/src/kernels/concat_webgpu.ts

```diff
@@ -17,23 +17,25 @@
 
 import {backend_util} from '@tensorflow/tfjs-core';
 
-import {computeDispatch} from '../webgpu_util';
+import {computeDispatch, flatDispatchLayout} from '../webgpu_util';
 import {WebGPUProgram} from './webgpu_program';
 
 export class ConcatProgram implements WebGPUProgram {
   outputShape: number[];
   userCode: string;
-  dispatchLayout: {x: number[], y: number[]};
+  dispatchLayout: {x: number[]};
   dispatch: [number, number, number];
   variableNames: string[];
+  workGroupSize: [number, number, number] = [64, 1, 1];
```

Comment:
Would you mind trying `workPerThread` > 1 to see whether that speeds things up even more?

Reply:
Done. But I didn't see much differences between these two situations. See my comments in https://github.com/tensorflow/tfjs/pull/2118#issuecomment-539433429